perfkitbenchmarker/linux_benchmarks/tensorflow_benchmark.py

```diff
@@ -0,0 +1,234 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run Tensorflow benchmarks (https://github.com/tensorflow/benchmarks)."""
+
+import logging
+import os
+import re
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.linux_packages import cuda_toolkit_8
+
+FLAGS = flags.FLAGS
+
+CUDA_TOOLKIT_INSTALL_DIR = cuda_toolkit_8.CUDA_TOOLKIT_INSTALL_DIR
+
+BENCHMARK_NAME = 'tensorflow'
+BENCHMARK_CONFIG = """
+tensorflow:
+  description: Runs Tensorflow Benchmark.
+  vm_groups:
+    default:
+      vm_spec:
+        GCP:
+          image: ubuntu-1604-xenial-v20170307
+          image_project: ubuntu-os-cloud
+          machine_type: n1-standard-4
+          gpu_type: k80
+          gpu_count: 1
+          zone: us-east1-d
+          boot_disk_size: 200
+        AWS:
+          image: ami-a9d276c9
+          machine_type: p2.xlarge
+          zone: us-west-2b
+          boot_disk_size: 200
+        Azure:
+          image: Canonical:UbuntuServer:16.04.0-LTS:latest
+          machine_type: Standard_NC6
+          zone: eastus
+"""
+
+flags.DEFINE_string('tf_model', 'vgg16', 'name of the model to run')
+flags.DEFINE_string('tf_data_name', 'imagenet',
```

Comment:
If this is in fact an enum, use flags.DEFINE_enum appropriately to make everyone's life easier